command/fs/locator.go

```diff
@@ -228,6 +238,11 @@ func (f *Locator) loadDirContents() {
 		return
 	}
 	for _, finfo := range contents {
+		if f.mod == Files && finfo.IsDir() {
+			continue
+		} else if f.mod == Dirs && !finfo.IsDir() {
+			continue
+		}
```

Comment:
I think this belongs in a method, e.g. `if f.mod.match(finfo) {` - it reads a bit cleaner and sidesteps an `else` statement.